src/test/components/Root.test.js

```diff
@@ -80,7 +80,8 @@ describe('app/AppViewRouter', function () {
     navigateToFromBrowserProfileLoadingPage();
     dispatch(waitingForProfileFromBrowser());
     expect(container.firstChild).toMatchSnapshot();
-    expect(retrieveProfileFromBrowser).toBeCalled();
+    // We do not check that retrieveProfileFromBrowser gets called, because the call
+    // happens asynchronously.
```

Comment:
Not sure. I didn't see any problems. If any problems crop up, we can take another look.

Personally, I don't like this test at all because of the way it mocks and stubs out a bunch of infrastructure. It relies on implementation details rather than making use of the regular APIs. It's bound to break if the implementation changes.